sdk/core/core-http/lib/restError.ts

```diff
@@ -21,6 +23,8 @@ export class RestError extends Error {
     response?: HttpOperationResponse
   ) {
     super(message);
+
+    this.name = "RestError";
```

Comment:
@xirzec , @chradek  - I was pretty sure we said this was supposed to be set to RestError but...it wasn't yet. Was that just a suggested design that hasn't yet been ratified?

Reply:
I am very surprised this isn't already a RestError...do errors currently just have the name `Error` then?

Reply:
Just tested again - the .name field is 'Error' even though the type of the class is RestError (ie, err instanceof RestError is true).

We don't have to figure that out for this PR - I moved to just testing for some RestError'ish fields instead, for now and I can file an issue about it if we want to talk about it more.